caper/cromwell_rest_api.py

```diff
@@ -247,82 +280,87 @@ def __request_get(self, endpoint, params=None):
         """
-        url = CromwellRestAPI.QUERY_URL.format(
-                ip=self._ip,
-                port=self._port) + endpoint
+        url = (
+            CromwellRestAPI.QUERY_URL.format(hostname=self._hostname, port=self._port)
+            + endpoint
+        )
         try:
             resp = requests.post(
-                url, files=manifest, auth=self._auth,
-                headers={'accept': 'application/json'})
-        except Exception as e:
-            logger.error(e)
-            sys.exit(1)
-
-        if resp.ok:
-            return resp.json()
-        else:
-            logger.error(
-                'POST: code={c}, contents={cont}, url={url}, manifest={m}'.format(
-                    c=resp.status_code, cont=resp.content, url=url, m=manifest))
-            return None
+                url,
+                files=manifest,
+                auth=self._auth,
+                headers={'accept': 'application/json'},
+            )
+            resp.raise_for_status()
+        except requests.exceptions.RequestException:
+            raise Exception(
+                'Failed to connect to Cromwell server. req=POST, url={url}'.format(
+                    url=url
+                )
+            ) from None
+        return resp.json()
 
     def __request_patch(self, endpoint, data):
         """POST request
 
         Returns:
             JSON response
         """
-        url = CromwellRestAPI.QUERY_URL.format(
-                ip=self._ip,
-                port=self._port) + endpoint
+        url = (
+            CromwellRestAPI.QUERY_URL.format(hostname=self._hostname, port=self._port)
+            + endpoint
+        )
         try:
             resp = requests.patch(
-                url, data=data, auth=self._auth,
-                headers={'accept': 'application/json',
-                         'content-type': 'application/json'})
-        except Exception as e:
-            logger.error(e)
-            sys.exit(1)
-
-        if resp.ok:
-            return resp.json()
-        else:
-            logger.error(
-                'PATCH: code={c}, contents={cont}, url={url}, json={j}'.format(
-                    c=resp.status_code, cont=resp.content, url=url, j=json))
-            return None
+                url,
+                data=data,
+                auth=self._auth,
+                headers={
+                    'accept': 'application/json',
+                    'content-type': 'application/json',
+                },
+            )
+            resp.raise_for_status()
+        except requests.exceptions.RequestException:
+            raise Exception(
+                'Failed to connect to Cromwell server. req=PATCH, url={url}'.format(
+                    url=url
+                )
+            ) from None
+        return resp.json()
 
     @staticmethod
     def __get_string_io_from_file(fname):
         with open(fname, 'r') as fp:
             return io.StringIO(fp.read())
 
+    @staticmethod
```

Comment:
Seems like you could use `self` for most of the `CromwellRestAPI` calls here.

Reply:
Removed these staticmethods. Please see above comments.